pkg/route/validation/validation.go

```diff
@@ -90,6 +101,33 @@ func validateRoute(route *routev1.Route, checkHostname bool) field.ErrorList {
 		result = append(result, err)
 	}
 
+	if len(route.Spec.HTTPHeaders.Actions.Response) != 0 || len(route.Spec.HTTPHeaders.Actions.Request) != 0 {
+		if route.Spec.TLS != nil && route.Spec.TLS.Termination == routev1.TLSTerminationPassthrough {
+			specTLS := field.NewPath(specTLSTerminationString)
+			result = append(result, field.NotSupported(specTLS, route.Spec.TLS.Termination, []string{"Only edge and re-encrypt routes are supported for providing customized headers."}))
+		}
+	}
+	if len(route.Spec.HTTPHeaders.Actions.Response) != 0 && len(route.Spec.HTTPHeaders.Actions.Response) > 20 {
+		result = append(result, field.Invalid(specPath.Child(specResponseString), route.Spec.HTTPHeaders.Actions.Response, "http response headers list can't exceed more than 20 items."))
+	}
+
+	if len(route.Spec.HTTPHeaders.Actions.Request) != 0 && len(route.Spec.HTTPHeaders.Actions.Request) > 20 {
+		result = append(result, field.Invalid(specPath.Child(specRequestString), route.Spec.HTTPHeaders.Actions.Request, "http request headers list can't exceed more than 20 items."))
+	}
+
+	if len(route.Spec.HTTPHeaders.Actions.Response) != 0 {
```

Comment:
Should we still validate the headers if the field is invalid due to having more than 20 items? If not, this should be an else clause, like so:
```go
if len(route.Spec.HTTPHeaders.Actions.Request) > 20 {
    // invalid
} else if len(route.Spec.HTTPHeaders.Actions.Request) != 0 {
    // do validation on each header
}
```

Reply:
Normally we would still check, because we want to append all the errors we can find before returning the answers.

Reply:
thanks Ryan! nice suggestion!

Reply:
https://github.com/openshift/library-go/commit/376ae6a3fee9be3ed62f1265c81cb8f535e8a2b5#diff-bfc6ba95cbc94986901ce1d0e3b74ee966499eab85a96e1d1ff763d2d74a7eeeR117-R135